### DISCUSSION TITLE
Ability to rename windows within sessions

### DISCUSSION THREAD

**Discussion by kissingfrogs2003** [sienori/Tab-Session-Manager#938] [Ideas]
I love this extension and there isn't much I would change. One feature suggestion though.... I really love being able to rename the saved "sessions" or groupings of windows BUT would also like to be able to rename the "Windows" within each of these through the same click & rename feature.

**So for example going from this setup of multiple saved sessions.....**
1. Things to Buy- Groceries- Beverages
2. Things to Buy- Groceries- Candy
3. Things to Buy- Furniture- Desks
4. Things to Buy- Furniture- Chairs
5. Things to Buy- Furniture- Bookshelves
6. Things to Buy- Furniture- Dining Table
7. Things to Buy- Storage Containers
8. Things to Buy- Crafts
9. Things to Buy- Shoes

**....to this setup only THREE sessions!**
1. Things to Buy- Groceries (2 windows, 4 tabs each)
     A. Beverages
     B. Candy
2. Things to Buy- Furniture (4 windows, 3 tabs each)
     A. Desks
     B. Chairs
     C. Bookshelves
     D. Dining Table
3. Things to Buy- Misc (3 windows, 2 tabs each)
     A. Storage Containers
     B. Crafts
     C. Shoes

I have tried to domy best within the current TSM capacities and have been using this workaround: 

- Start a Google search with the "title" of the window (e.g. Things to Buy- Groceries- Beverages)
- Make sure that this the 1st tab in the browser
- Make sure this search is the ACTIVE tab when adding the window to TSM.

I have included a screenshot of how this workaround has allowed for "groupings" and "labeled windows" within particular sessions. And while this works well enough, it would be AWESOME if there was a more elegant or simpler way to have the windows able to be renamed as easily as we can with sessions! 
<img src="https://user-images.githubusercontent.com/98179211/150574805-4691b5af-c767-4cdf-bdc9-ab95932a729f.png" width="600" height="450">

**Comment by rediffusion** [comment]
Hi bro! 🙂
I see we’re concerned about the same thing.

A similar issue is [here](https://github.com/sienori/Tab-Session-Manager/discussions/996).
This is [my method](https://github.com/sienori/Tab-Session-Manager/discussions/996#discussioncomment-2827532) how working. 

Thank you for your interest.
